### PR TITLE
Datadog config

### LINKED ISSUE #305: Notify deploys to Datadog
## Description

We need to know when deployments happen so that we can correlate issues (or not). While a mem and cpu usage spike is fine in a deploy it is not in other circumstances.

Visually knowing this information in all metrics graphs would allow correlating things at a glance.

## Proposed solution

Investigate what's the way to do this with Datadog and implement it. It shouldn't be hard.

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Closes #305 

This PR: 
- improves our current Datadog settings
- notifies Datadog on a successful deploy or provision
- enables collection of postgresql stats and logs through the Datadog agent
- also enables nginx stats and logging to Datadog

Screenshot-2019-04-25-13-21:57

:warning: Provisioning the new postgres stats and logging features to a server requires adding or updating these variables in `secrets.yml`:

```ruby
datadog_db_password: <secure_password_goes_here> # use $ openssl rand -hex 128

db_integrations:
  - { user: datadog, state: present, password: "{{ datadog_db_password }}" }
```
And if you havn't done it already, you'll also need to run `playbooks/db_integrations.yml` first to create the database user and grant them permissions.

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
I've tested this extensively in Staging, and it's now live on UK production as well :+1:

File: roles/datadog/tasks/pg_stats.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            port: 5432
+            username: datadog
+            password: "{{ datadog_db_password }}"
+            dbname: "{{ db }}"
+            tags:
+              - "host:{{ inventory_hostname }}"
+            collect_activity_metrics: true
+        logs:
+          - type: file
+            path: /var/log/pg_log/postgresql-*.log
```
Comment (sauloperez) [inline roles/datadog/tasks/pg_stats.yml]: Are we sure about that? As far as I read setting pg_log as the log destination stores them within the Postgres installation directory and not in /var/log.

Comment (sauloperez) [reply roles/datadog/tasks/pg_stats.yml]: Check `log_directory` in https://github.com/openfoodfoundation/ofn-install/pull/410/files

Comment (Matt-Yorkley) [reply roles/datadog/tasks/pg_stats.yml]: We havn't had any `stderr` output form UK prod yet, so we don't know... I'll check it again though

Comment (Matt-Yorkley) [reply roles/datadog/tasks/pg_stats.yml]: https://docs.datadoghq.com/integrations/postgres/#log-collection

Comment (Matt-Yorkley) [reply roles/datadog/tasks/pg_stats.yml]: You're right, so I've opted to go with the suggestions you quoted from here: https://logdna.com/postgresql-logging-guide/

**Comment by Matt-Yorkley** [comment]
@sauloperez this turned out to be a lot more complicated than I originally thought. I decided to only apply the logging to servers that will use it, because the logs are massive and we will only be collecting them on 3 servers. I also switched the way we apply the logging config to use the `/conf.d` pattern, because it's difficult to cleanly change or remove things that have written lots of lines into the main config file in a haphazard way.

**Comment by Matt-Yorkley** [comment]
We can get really nice Rails logs collected in Datadog as well and I think it would be really useful, but we need to discuss signing up for the log service to get log retention > 1 day. It's an extra $2-3 per million logs. If we do that we should probably look at raising the log level above `info` for postgres logs.

**Comment by Matt-Yorkley** [comment]
I've reprovisioned UK prod :+1:

**Pull request merged**